### DISCUSSION TITLE
Close PR even with changes

### DISCUSSION THREAD

**Discussion by JavierSegoviaCordoba** [renovatebot/renovate#16593] [Request Help] [answered]
Some times I am working on a feature and I merge manually and push to main, so the renovate PR it is no longer necessary.

The problem is that I have a workflow which add the dependency update to the changelog so Renovate checks that its PR is altered and can't close it.

Is there a option to close unnecessary PRs even if they are changed?

A similar use case, I have a PR which is changed by adding it to the changelog and a new version appears, I have to click manually to update it.

This desired behavior was working before, not sure since which version started to happen.

An specific problem that can appear and it was not happening is that the PR has to be updated only if a new version appears, not every time it is changed.

**Comment by viceice** [comment, marked as answer]
try this setting

https://docs.renovatebot.com/configuration-options/#gitignoredauthors

**Reply by viceice** [reply]
fixed wrong link

**Reply by JavierSegoviaCordoba** [reply]
Wow, I have checked that I have it already 🤔 

```json
{
  "extends": [
    "config:base"
  ],
  "assigneesFromCodeOwners": true,
  "commitBodyTable": true,
  "gitIgnoredAuthors": [
    "javiersegoviacordoba@gmail.com",
    "javier@segoviacordoba.com"
  ]
}
```

cc @rarkins (I don't know if replying to you notifies Rarkins too)

Sample repo: https://github.com/JavierSegoviaCordoba/semver-gradle-plugin/issues/2
Renovate json: https://github.com/JavierSegoviaCordoba/semver-gradle-plugin/blob/main/.github/renovate.json

**Reply by rarkins** [reply]
Check the debug logs

**Reply by JavierSegoviaCordoba** [reply]
```
DEBUG: Last commit author does not match git author email - branch has been modified(branch="renovate/hubdle")
{
  "branchName": "renovate/hubdle",
  "lastAuthor": "29139614+renovate[bot]@users.noreply.github.com",
  "gitAuthorEmail": "bot@renovateapp.com"
}
DEBUG: Found existing branch PR(branch="renovate/hubdle")
DEBUG: PR has been edited(branch="renovate/hubdle")
{
  "prNo": 72
}
DEBUG: Updating existing PR to indicate that rebasing is not possible(branch="renovate/hubdle")
DEBUG: updatePr(72, Update dependency com.javiersc.hubdle to v0.2.0-alpha.21, body)(branch="renovate/hubdle")
DEBUG: PR updated(branch="renovate/hubdle")
{
  "pr": 72
}
```

**Reply by JavierSegoviaCordoba** [reply]
I am amending the commit (in the past too), maybe something has changed so Renovate is not able to see that the force push comes from my account token, so my emails don't work, and I have to add `bot@renovateapp.com` as ignored?

**Reply by JavierSegoviaCordoba** [reply]
As you can see here:

[JavierSegoviaCordoba](https://github.com/JavierSegoviaCordoba) [force-pushed](https://github.com/JavierSegoviaCordoba/semver-gradle-plugin/compare/d3802bd5f8a983276ff3bb7ee35315d8a9916160..556b2ee03634846873d34bfc9b07da9a6ec23922) the renovate/hubdle branch from [d3802bd ](https://github.com/JavierSegoviaCordoba/semver-gradle-plugin/commit/d3802bd5f8a983276ff3bb7ee35315d8a9916160)to [556b2ee ](https://github.com/JavierSegoviaCordoba/semver-gradle-plugin/commit/556b2ee03634846873d34bfc9b07da9a6ec23922)[3 days ago](https://github.com/JavierSegoviaCordoba/semver-gradle-plugin/pull/72#event-6978317156)

**Reply by rarkins** [reply]
Are you doing the push with Renovate's git author? It looks like that:

<img width="511" alt="image" src="https://user-images.githubusercontent.com/6311784/179214161-839b8f1f-8716-4272-9de8-8e8ec2eeeb72.png">

In the app we have this:

```
    gitAuthor: 'Renovate Bot <bot@renovateapp.com>',
    gitIgnoredAuthors: ['29139614+renovate[bot]@users.noreply.github.com'],
```

I will probably *swap* these soon, but I think the problem right now is that your own definition of `gitIgnoredAuthors` overrides the app's default.

You could also consider skipping the force push and instead adding a commit with one of your own ignored authors.

**Reply by JavierSegoviaCordoba** [reply]
```
- name: Commit and push
        run: |
          git config --global user.email "action@github.com"
          git config --global user.name "GitHub Action"
          git diff
          git add .
          git commit --amend --no-edit
          git push --force
```

Curiously, before the email associated to the token has the preference, looks like now it is the renovate or the github email I put there 🤔

**Reply by JavierSegoviaCordoba** [reply]
I want to keep the amend, should I add the renovate email then, right?

**Reply by JavierSegoviaCordoba** [reply]
Renovate checks the `renovate.json` in `main` or the `renovate.json` when the PR was created? I have updated it on main to add those two emails but Renovate is not able to regenerate them because they are edited.

**Reply by rarkins** [reply]
renovate.json in main

**Reply by JavierSegoviaCordoba** [reply]
This should be updated, it isn't: https://github.com/JavierSegoviaCordoba/semver-gradle-plugin/pull/73

```
DEBUG: latest repository commit(branch="renovate/gradle-7.x")
{
  "latestCommit": {
    "hash": "4e05d5be75c95236c7e33057b5c299df46e70716",
    "date": "2022-07-15T11:46:30+00:00",
    "message": "Update dependency com.javiersc.hubdle to v0.2.0-alpha.25",
    "refs": "HEAD -> main, origin/main, origin/HEAD",
    "body": "| datasource | package                                               | from           | to             |\n| ---------- | ----------------------------------------------------- | -------------- | -------------- |\n| maven      | com.javiersc.hubdle:com.javiersc.hubdle.gradle.plugin | 0.2.0-alpha.20 | 0.2.0-alpha.25 |\n",
    "author_name": "renovate[bot]",
    "author_email": "29139614+renovate[bot]@users.noreply.github.com"
  }
}
DEBUG: Last commit author does not match git author email - branch has been modified(branch="renovate/gradle-7.x")
{
  "branchName": "renovate/gradle-7.x",
  "lastAuthor": "29139614+renovate[bot]@users.noreply.github.com",
  "gitAuthorEmail": "bot@renovateapp.com"
}
DEBUG: Found existing branch PR(branch="renovate/gradle-7.x")
DEBUG: PR has been edited(branch="renovate/gradle-7.x")
{
  "prNo": 73
}
```

I guess this `bot@renovateapp.com` email is not valid and I have to add the other one (`29139614+renovate[bot]@users.noreply.github.com`)?

As you said, you are going to change that email, which is going to be the final one so I can put it now and forget?

**Reply by rarkins** [reply]
Longer term it will be `29139614+renovate[bot]@users.noreply.github.com` for everything

**Comment by rarkins** [comment]
It sounds like you should include your other automated service's git author in gitIgnoredAuthors: https://docs.renovatebot.com/configuration-options/

That way the branch won't be declared as modified and can receive branch updates or be auto closed

**Reply by JavierSegoviaCordoba** [reply]
Thank you both :)
